src/graphics/rc_texture.rs

```diff
@@ -127,6 +127,18 @@ impl RcTexture {
         self.texture.borrow_mut().create(width, height)
     }
 
+    /// Get the source texture of an [`RcTexture`]
+    ///
+    /// It let's you temporarily borrow the [`Texture`] being held by an [`RcTexture`].
+    /// This may be useful for many things, for example, using vertex arrays with
+    /// [`RcTexture`].
+    ///
+    /// Returns a [`Texture`]
+    #[must_use]
+    pub fn raw_texture(&self) -> &Texture {
+        unsafe { &*self.texture.as_ptr() }
```

Comment:
I'm not 100% sure if it's sound to just get an immutable reference out of a RefCell like this, but right now I can't find any example where it causes undefined behavior.
But since rc_texture is your domain, I'll just let you reign free.